### PR TITLE
Release/v4.2.0

### PR DESCRIPTION
**Pull request by matthew-dean**

### CODE REVIEW

**Comment by matthew-dean** [comment]
@iChenLei Crap, can you help figure out the reason for these failures? This should just include the [PR for container queries](https://github.com/less/less.js/pull/3811), yet even though those tests were passing, these fail. I released 4.2.0, so it would be good to figure out what's wrong. I ran tests locally and they succeed, so what's different about CI?

**Comment by matthew-dean** [comment]
@iChenLei I'm inclined to merge this, since this is what was _released_, but we definitely need to figure out the error reasons.

**Comment by matthew-dean** [comment]
@iChenLei Another question -- how are you managing changelogs now? Manual updates?

File: package.json [hunk 1]
```diff
@@ -27,7 +26,7 @@
   },
   "devDependencies": {
     "github-changes": "^1.1.2",
-    "lerna": "^3.22.1",
+    "lerna": "^7.1.4",
```
Comment (iChenLei) [inline package.json:29]: I think it's very likely due to the lerna upgrade.

**Comment by iChenLei** [comment]
> @iChenLei Crap, can you help figure out the reason for these failures? This should just include the [PR for container queries](https://github.com/less/less.js/pull/3811), yet even though those tests were passing, these fail. I released 4.2.0, so it would be good to figure out what's wrong. I ran tests locally and they succeed, so what's different about CI?

https://github.com/less/less.js/blob/708da87ca51d62e89c77d68e6ccc6d9ad661f271/package.json#L29

Sir, you upgrade the lerna from v3 to v7 (these major versions include some breaking changes, such as [lerna bootstrap deprecated](https://lerna.js.org/docs/legacy-package-management#migrating-from-lerna-bootstrap-lerna-add-and-lerna-link-in-lerna-v7-and-later)).

| lerna v3 | lerna v7 |
|:---:|:---:|
|   |  |

In lerna v3, the lerna cli install deps and Symlinking packages.
```bash
lerna notice cli v3.22.1
lerna info ci enabled
lerna info Bootstrapping 3 packages
lerna info Installing external dependencies
lerna info Symlinking packages and binaries
lerna success Bootstrapped 3 packages
```
but lerna v7 not, so we encounter follow issues:
```bash
Fatal error: Unable to find local grunt.
```
I fork your branch and add some changes:
https://github.com/less/less.js/blob/d63503ef9d3d081185c278f4457cd18534e72205/package.json#L3
But it still has some issues: https://github.com/less/less.js/actions/runs/5774396195/job/15651297281#step:6:148

**Comment by iChenLei** [comment]
> @iChenLei I'm inclined to merge this, since this is what was _released_, but we definitely need to figure out the error reasons.

options 1: downgrade lerna and releases v4.2.0
options 2: fix lerna upgrade issue

**Comment by iChenLei** [comment]
> @iChenLei Another question -- how are you managing changelogs now? Manual updates?

Yes，Manual updates

**Comment by matthew-dean** [comment]
@iChenLei ohhh ok, maybe it's resolving something locally which can't be resolved in CI, thanks for investigating.

**Comment by matthew-dean** [comment]
@iChenLei The Windows test is permanently stuck, no matter how often it's run. It never runs the unit rest step. Can you investigate? Otherwise I've fixed the other tests. Also would you be able to update the change log?

**Comment by matthew-dean** [comment]
Merging because all the other platforms work and no explicit errors are given. @iChenLei I'd be grateful if you could help figure out the Windows issue.

**Pull request merged**